universal-application-tool-0.0.1/app/views/BaseHtmlView.java

```diff
@@ -21,17 +21,18 @@ protected Content htmlContent(DomContent... domContents) {
   }
 
   protected Tag textField(String fieldName, String labelText) {
-    return label(text(labelText), input().withType("text").withName(fieldName))
+    return label(text(labelText), input().withType("text").withName(fieldName).withId(fieldName))
         .attr("for", fieldName);
   }
 
   protected Tag passwordField(String fieldName, String labelText) {
-    return label(text(labelText), input().withType("password").withName(fieldName))
+    return label(
+            text(labelText), input().withType("password").withName(fieldName).withId(fieldName))
         .attr("for", fieldName);
   }
 
   protected Tag submitButton(String textContents) {
-    return input().withType("submit").withValue(textContents);
+    return input().withType("submit").withId("submit").withValue(textContents);
```

Comment:
(same issue as above)

Issue (blocking): names and element IDs have different behavior and we should not encourage them defaulting them to the same value. It's common for multiple elements on the page or even within the same

(as is the case for checkboxes) to have the same name, but no two elements on the page should have the same ID.
Suggestion: have a second version of this method that allows specifying the ID.

Reply:
Done as above (textfield also)